syft/format/common/spdxhelpers/to_format_model.go

```diff
@@ -713,8 +714,8 @@ func toFileTypes(metadata *file.Metadata) (ty []string) {
 	return ty
 }
 
-// other licenses are for licenses from the pkg.Package that do not have an SPDXExpression
-// field. The spdxexpression field is only filled given a validated Value field.
+// other licenses are for licenses from the pkg.Package that do not have a valid SPDX Expression
+// OR are an expression that is a single `License-Ref-*`
```

Comment:
I've seen multiple references in comments and PR descriptions about `License-Ref`, but the code handles `LicenseRef`... the code looks consistent with what I think should happen, is this just an inconsistency in docs? Or a case I'm missing?

Reply:
The docs changed here because `The spdxexpression field is only filled given a validated Value field` is a case we must handle when we discover `License-Ref-*` as license metadata (one we did NOT create).

`License-Ref-*`, when discovered by syft, will validate as a correct SPDXExpression. Our previous logic would only act on a license in `toOtherLicenses` if `SPDXExpression == ""`.

This was missing the case where `SPDXExpression=="Licenses-Ref-*"` which would require an entry to be created for `spdx.OtherLicenses`